org.eclipse.draw2d.tests/src/org/eclipse/draw2d/test/swtbot/AbstractSWTBotTests.java

```diff
@@ -33,31 +34,39 @@
 
 import org.eclipse.draw2d.FigureCanvas;
 import org.eclipse.draw2d.IFigure;
+import org.eclipse.draw2d.test.swtbot.AbstractSWTBotTests.SWTBotExtension;
 import org.eclipse.draw2d.test.utils.Snippet;
 
-import org.junit.Rule;
-import org.junit.rules.TestRule;
-import org.junit.runners.model.Statement;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.InvocationInterceptor.Invocation;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
 
 /**
  * To be able to execute these tests in the Eclipse IDE, the tests must
  * <b>NOT</b> be run in the UI thread.
  */
 @SuppressWarnings("nls")
+@ExtendWith(SWTBotExtension.class)
 public abstract class AbstractSWTBotTests {
 	protected SWTBotCanvas bot;
 	protected FigureCanvas canvas;
 	protected IFigure root;
 
-	@Rule
-	public TestRule rule = (base, description) -> new Statement() {
+	public static class SWTBotExtension implements InvocationInterceptor {
 		@Override
-		public void evaluate() throws Throwable {
-			Snippet annotation = description.getAnnotation(Snippet.class);
-			Objects.requireNonNull(annotation, "Test is missing @Snippet annotation."); //$NON-NLS-1$
-			doTest(annotation, base);
+		public void interceptTestMethod(Invocation<Void> invocation,
+				ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext)
+				throws Throwable {
+			Object target = invocationContext.getTarget().orElse(null);
+			if (target instanceof AbstractSWTBotTests testObject) {
+				Snippet annotation = invocationContext.getExecutable().getAnnotation(Snippet.class);
+				Objects.requireNonNull(annotation, "Test is missing @Snippet annotation."); //$NON-NLS-1$
+				testObject.doTest(annotation, invocation);
+			}
```

Comment:
There's probably a better way to do this but at the end of the day, I'm not overly familiar with the JUnit Extension API. In order to execute the SWTBot tests, we need to execute some additional code that creates and disposes the widget covered by the test...

Reply:
I'm a big fan of if it works and improves the existing situation. So I would go withit.